iree/test/e2e/linalg_ext_ops/CMakeLists.txt

```diff
@@ -57,6 +60,7 @@ iree_check_single_backend_test_suite(
     check_vulkan-spirv_vulkan
   SRCS
     "reverse.mlir"
+    "top-k.mlir"
```

Comment:
This appears to be failing on post-submit on Android (Pixel 6 and Moto Edge X30):

https://buildkite.com/iree/iree-test-android/builds/6253#56f69286-88d7-459d-b8d3-5803755ab4bb

```
  9/467 Test   #9: iree/test/e2e/linalg_ext_ops/check_vulkan-spirv_vulkan_top-k.mlir_on_android_device 
...
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from module
[ RUN      ] module.topk_1d_dim0_max
[       OK ] module.topk_1d_dim0_max (74 ms)
[ RUN      ] module.topk_1d_dim0_min
[       OK ] module.topk_1d_dim0_min (63 ms)
[ RUN      ] module.topk_2d_dim1_max
../runtime/src/iree/modules/check/module.cc:352: Failure
Failed
Expected near equality of these values. Contents does not match.
  lhs:
    2x3xf32=[10 9 8][12 11 10]
  rhs:
    2x3xf32=[6 5 4][12 11 10]
../runtime/src/iree/modules/check/module.cc:270: Failure
Failed
Expected equality of these values. Contents does not match.
  lhs:
    2x3xi32=[0 0 0][11 10 9]
  rhs:
    2x3xi32=[5 4 3][11 10 9]
[  FAILED  ] module.topk_2d_dim1_max, where GetParam() = 2 (54 ms)
[ RUN      ] module.topk_2d_dim1_inverted_max
[       OK ] module.topk_2d_dim1_inverted_max (46 ms)
[----------] 4 tests from module (239 ms total)
 
[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (239 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] module.topk_2d_dim1_max, where GetParam() = 2
 
 1 FAILED TEST
Test failed
OK
```

Reply:
@KoolJBlack To debug it, we may mark all loops as reduction loops. If it passes, the issue could be in tiling method.

Reply:
Interesting. I'm about to go to bed but I'll take a look at this in first thing in the morning. Curios why this is failing on just one phone only post submit. Is there anything particularly different about the test running on this phone (and why it doesn't affect the other phone)?

Reply:
I think this is happening on all phones. Other phones might just have other errors

Reply:
Given that this passes on CPU, my bet is that the second level of tile and distribute on SpIRV might be off.

Reply:
To clarify, by "second level of tiling" do you mean the second dim of the tile size? 

To diagnose this issue, can I trigger the second level of tiling using a specific __internal_linalg_transform__ in `tiling.mlir`? Currently I have the `inner_reduce_input` tag -- looking through the test its running with 

```
      context, linalg::LinalgTilingOptions().setTileSizes({10, 0, 0}),
```

Which I'm guessing is just first level tiling?

Reply:
We apply the other tile+distribute in GPU backends. IIRC, that would distribute workloads at thread level. Here is the example in SPIR-V: https://github.com/google/iree/blob/main/compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTileAndDistribute.cpp

This could be the issue that we don't set configurations in KernelConfigs for CUDA and SPIR-V backend. That would make the second level of tile and distribute fail.

I looked into the past PR I submitted. Before having proper configurations, we should set all the loops as reduction loop. (See l.544 in [iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp](https://github.com/google/iree/pull/7067/files#diff-4d318a7e0d67c901fa233185a9d48d9b177692e71dc8b1bd30fd0903f6581e39)) This will make all computation sequentially. 

After adding proper. configurations, we can mark them back to parallel loops.

This is the PR that adding tileAndDistribute for FFT: https://github.com/google/iree/pull/7067

Kojo, maybe we can mark all iteration types as reduction and see what's happening? If it works for all the backends, the issue could be either in tiling method or bad configuration.

Reply:
What Hanhan said :)  (Sorry was responding from my phone, so was abrupt, and should have made the terminology clearer). There is one level of tile and distribute to workgroups (i.e. thread blocks) and another level to threads (i.e. threads). I guess the SPIR-V side doesnt set a default for the second level. Still should serialize the computation on one thread and produce correct results. Incorrect results is surprising. 
@antiagainst something to fix more structurally.

@KoolJBlack I would just ignore the Vulkan side for now... If the tests pass for CUDA, then things are probably OK to move ahead...

Reply:
oh, I was not aware that it passes for CUDA. If it only fails for vulkan, let's file an issue and move ahead.

Reply:
Just an update: 

I tried setting all the loops to reduction, using the `getIterationDomain()` method, but this fails to compile. The error complains about a missing implementation of tiling for a dimension. Mehash said this is likely caused by tiling being applied to at least one of the dimension yet none of them can be tiled. Messing around, i couldn't get around this issue, I'd need to check what tile sizes are being applied which would require deconstructing the e2e tests quite a bit more. 

I did however verify that these tests do work on both the cuda and vulkan backends on my own machine. This is a vulkan run: https://gist.github.com/KoolJBlack/29031ab74d6fa5d36878ec2bf2d240f4

Given that I'm not seeing this failure anywhere else and the vulkan test was disabled for now I'm going to leave this issue until it pops up again beyond our lab phones.